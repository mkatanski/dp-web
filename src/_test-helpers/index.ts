export * from "./withResponsiveContext";
export * from "./withThemeProvider";
