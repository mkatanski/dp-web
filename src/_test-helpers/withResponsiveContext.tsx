import React from "react";
import { Context as ResponsiveContext } from "react-responsive";
import { Breakpoints } from "config/breakpoints";

export const withMobileSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.mobile }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withTabletSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.tablet }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withDesktopSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.desktop }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withLargeDesktopSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.largeDesktop }}>
    {Component}
  </ResponsiveContext.Provider>
);
