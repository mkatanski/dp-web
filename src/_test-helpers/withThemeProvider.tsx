import React from "react";
import { ThemeProvider as DefaultThemeProvider } from "lib/theme/ThemeProvider";
import { StyledTheme, Theme } from "styles/theme/_types";
import { createMuiTheme } from "@material-ui/core";

const testTheme: StyledTheme = {
  primary: "primary",
  secondary: "secondary",
  textPrimary: "textPrimary",
  textSecondary: "textSecondary",
  textInverse: "textInverse"
};

const muiTheme = createMuiTheme({
  direction: "ltr"
});

const theme: Theme = Object.assign(testTheme, muiTheme);

export const withThemeProvider = (Component: React.ReactNode) => (
  <DefaultThemeProvider theme={theme}>{Component}</DefaultThemeProvider>
);
