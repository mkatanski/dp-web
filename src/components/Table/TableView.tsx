import React from "react";
import styled from "styled-components";
import { Table } from "@material-ui/core";
import { omit } from "utils/helpers";

export type TableViewProps = {
  striped?: boolean;
};

export const TableView = styled<React.ComponentType<TableViewProps>>(props => (
  <Table stickyHeader {...omit(props, "striped")} />
))`
  position: absolute;

  th {
    padding: ${({ theme }) => theme.spacing(1, 2)};
  }

  td {
    padding: ${({ theme }) => theme.spacing(2, 2)};
  }

  tr:nth-child(even) {
    background: ${({ theme, striped }) =>
      striped ? theme.palette.grey[50] : "inherit"};
  }
`;

TableView.displayName = "TableView";
