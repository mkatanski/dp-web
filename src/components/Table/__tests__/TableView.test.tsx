import React from "react";
import TestRenderer from "react-test-renderer";
import "jest-styled-components";

/* Uncomment below if needed */
import { withThemeProvider } from "_test-helpers";
// import { render, getByTestId } from "@testing-library/react";

import { TableView } from "../TableView";

describe("components", () => {
  describe("Table", () => {
    describe("TableView", () => {
      it("should render TableView component", () => {
        const tree = TestRenderer.create(withThemeProvider(<TableView />));
        expect(tree).toMatchSnapshot();
      });

      it("should render TableView component with children", () => {
        const tree = TestRenderer.create(
          withThemeProvider(<TableView>child</TableView>)
        );
        expect(tree).toMatchSnapshot();
      });

      it("should render TableView component striped", () => {
        const tree = TestRenderer.create(
          withThemeProvider(<TableView striped />)
        );
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
