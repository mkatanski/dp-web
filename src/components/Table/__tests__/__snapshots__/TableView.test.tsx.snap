// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Table TableView should render TableView component 1`] = `
.c0 {
  position: absolute;
}

.c0 th {
  padding: 8px 16px;
}

.c0 td {
  padding: 2px 16px;
}

.c0 tr:nth-child(even) {
  background: inherit;
}

<table
  className="c0"
  data-props={
    Object {
      "stickyHeader": true,
    }
  }
  id="MUI.Table"
/>
`;

exports[`components Table TableView should render TableView component striped 1`] = `
.c0 {
  position: absolute;
}

.c0 th {
  padding: 8px 16px;
}

.c0 td {
  padding: 2px 16px;
}

.c0 tr:nth-child(even) {
  background: tableRowBgEven;
}

<table
  className="c0"
  data-props={
    Object {
      "stickyHeader": true,
    }
  }
  id="MUI.Table"
/>
`;

exports[`components Table TableView should render TableView component with children 1`] = `
.c0 {
  position: absolute;
}

.c0 th {
  padding: 8px 16px;
}

.c0 td {
  padding: 2px 16px;
}

.c0 tr:nth-child(even) {
  background: inherit;
}

<table
  className="c0"
  data-props={
    Object {
      "stickyHeader": true,
    }
  }
  id="MUI.Table"
>
  child
</table>
`;
